### PR TITLE
Additional fixes

### PR DESCRIPTION
**Pull request by SofiaChar**
1. New docker image `valohai/llm-toolkit`
2. Fix error in data-prep
3. Delete` viggo.py` file from inputs to data-prep step, now load without HF script
4. Minor change in finetuning parameters
5. Add `prepare_prompt` method to inference to have the proper input to the model when inferencing.

### CODE REVIEW

**Comment by SofiaChar** [comment]
@tokkoro Can i merge this?

File: inference-mistral.py [hunk 1]
```diff
@@ -3,6 +3,7 @@
 
 import torch
 import transformers
+import datasets
```
Comment (tokkoro) [inline inference-mistral.py]: Is this used somehow?

You should install the [pre-commit](https://pypi.org/project/pre-commit/) hooks to run [ruff](https://github.com/astral-sh/ruff) locally before committing.  It would catch these kind of issues and fix them for you when you are about to commit new code.

To install:
```
pip install pre-commit
pre-commit install
```
then you can run `ruff .` to see all issues or `ruff . --fix` to fix them. Or run them via pre-commit without commit by `pre-commit run --all-files`. The configs for pre-commit are in https://github.com/valohai/mistral-example/blob/main/.pre-commit-config.yaml

File: inference-mistral.py [hunk 2]
```diff
@@ -52,11 +64,12 @@ def main():
...
     parser = argparse.ArgumentParser(description='Fine-tuned Model Inference')
     # fmt: off
-    parser.add_argument('--base_mistral_model', type=str, default='mistralai/Mistral-7B-v0.1', help='Base mistral from hugging face')
+    parser.add_argument('--base_mistral_model', type=str, default='mistralai/Mistral-7B-v0.1',
+                        help='Base mistral from hugging face')
     parser.add_argument('--checkpoint_path', default='/valohai/inputs/finetuned-checkpoint/')
     parser.add_argument('--max_tokens', type=int, default=305, help='Maximum number of tokens in response')
     parser.add_argument('--model_path', default='/valohai/inputs/model-base/')
-    parser.add_argument('--prompt', type=str, required=True, help='Input prompt for text generation')
+    parser.add_argument('--prompt', type=str, help='Input prompt for text generation')
```
Comment (akx) [inline inference-mistral.py]: The prompt _is_ required, though, isn't it? You can't infer without one, and passing `None` would fail.

File: inference-mistral.py [hunk 3]
```diff
@@ -52,11 +64,12 @@ def main():
     logging.basicConfig(level=logging.INFO)
     parser = argparse.ArgumentParser(description='Fine-tuned Model Inference')
     # fmt: off
-    parser.add_argument('--base_mistral_model', type=str, default='mistralai/Mistral-7B-v0.1', help='Base mistral from hugging face')
+    parser.add_argument('--base_mistral_model', type=str, default='mistralai/Mistral-7B-v0.1',
+                        help='Base mistral from hugging face')
```
Comment (akx) [inline inference-mistral.py]: Unnecessary change here.

File: inference-mistral.py [hunk 4]
```diff
@@ -27,13 +28,24 @@ def __init__(self, model_path: str, checkpoint_path: str) -> None:
...
     def generate_response(self, prompt: str, max_tokens: int = 50) -> str:
-        inputs = self.tokenizer(prompt, return_tensors='pt')
+        inputs = self.prepare_prompt(prompt)
         with torch.no_grad():
             logger.info('Generating up to %d tokens...', max_tokens)
             outputs = self.ft_model.generate(**inputs, max_length=max_tokens, pad_token_id=2)
         logger.info('Decoding...')
         return self.tokenizer.decode(outputs[0], skip_special_tokens=True)
 
+    def prepare_prompt(self, prompt):
```
Comment (akx) [inline inference-mistral.py:37]:
```suggestion
    def prepare_prompt(self, prompt: str):
```

I'm not sure what the return type is since we're splatting it with `**` to `self.ft_model.generate`...

File: data-preprocess.py [hunk 5]
```diff
@@ -12,12 +12,12 @@
...
-        self.data_path = args.data_path or valohai.inputs('dataset').path()
+        self.data_path = args.data_path
         self.model_max_length = args.model_max_length
         self.tokenizer = args.tokenizer
-        self.train_dataset = load_dataset(self.data_path, split='train')
-        self.eval_dataset = load_dataset(self.data_path, split='validation')
-        self.test_dataset = load_dataset(self.data_path, split='test')
+        self.train_dataset = load_dataset('csv', data_files=valohai.inputs('dataset').path('train.csv'))
+        self.eval_dataset = load_dataset('csv', data_files=valohai.inputs('dataset').path('validation.csv'))
+        self.test_dataset = load_dataset('csv', data_files=valohai.inputs('dataset').path('test.csv'))
```
Comment (akx) [inline data-preprocess.py]: Don't these changes mean that the `data_path` argument isn't used anymore? If so, it should be removed altogether.

However, I think it would be better to allow it, and compose these paths based on it (and default it to the `dataset` input directory)?

Maybe more broadly – what's the reason behind this change? Doesn't this work like the original code did?

**Review by akx — CHANGES_REQUESTED** [review]

File: data-preprocess.py (hunk 5 above)
Comment (SofiaChar) [reply data-preprocess.py]: In the original code the dataset input was pointing to `viggo.py` file, which loaded the data from HF for us. We decided that it's not the intuitive way to load the data, and it is better to load from csv. When loading from csv you should pass the path to the exact file. I don’t see how we can have one `data_path`.
Maybe it will look better if we have three inputs like: train_data, test_data, val_data?
Or I delete `data_path` arg for good?

File: inference-mistral.py (hunk 4 above)
Comment (SofiaChar) [reply inference-mistral.py:37]: It returns something like `transformers.tokenization_utils_base.BatchEncoding`. I prefer not to specify the type here

File: data-preprocess.py (hunk 5 above)
Comment (akx) [reply data-preprocess.py]: > When loading from csv you should pass the path to the exact file. I don’t see how we can have one `data_path`.

It'd be the path to the directory with the files:

```suggestion
        self.data_path = args.data_path or valohai.inputs('dataset').path()
        self.model_max_length = args.model_max_length
        self.tokenizer = args.tokenizer
        self.train_dataset = load_dataset('csv', data_files=os.path.join(self.data_path, 'train.csv'))
        self.eval_dataset = load_dataset('csv', data_files=os.path.join(self.data_path, 'validation.csv'))
        self.test_dataset = load_dataset('csv', data_files=os.path.join(self.data_path, 'test.csv'))
```

**Review by akx — APPROVED** [review]
